# HG changeset patch
# User alta88@fixall.com
# Date 1729015920 -7200
# Parent  e78637cd6fd9f488291440aa305c3c376fe11939
Fix feed vandalization, Part 9 - make feed drag/drop onto folder pane work again.

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3143,8 +3143,22 @@ var folderPane = {
         }
       }
       event.dataTransfer.dropEffect =
         systemDropEffect == "copy" ? "copy" : "move";
+    } else if (
+      targetFolder.server.type == "rss" &&
+      (types.includes("text/x-moz-url-data") ||
+        types.includes("text/x-moz-url"))
+    ) {
+      // Allow subscribing to feeds by dragging an url to a feed account.
+      if (
+        targetFolder.isSpecialFolder(Ci.nsMsgFolderFlags.Trash, true) ||
+        event.dataTransfer.items.length != 1 ||
+        !FeedUtils.getFeedUriFromDataTransfer(event.dataTransfer)
+      ) {
+        return;
+      }
+      event.dataTransfer.dropEffect = "link";
     } else if (types.includes("application/x-moz-file")) {
       if (targetFolder.isServer || !targetFolder.canFileMessages) {
         return;
       }
@@ -3171,22 +3185,8 @@ var folderPane = {
           return;
         }
       }
       event.dataTransfer.dropEffect = "move";
-    } else if (
-      types.includes("text/x-moz-url-data") ||
-      types.includes("text/x-moz-url")
-    ) {
-      // Allow subscribing to feeds by dragging an url to a feed account.
-      if (
-        targetFolder.server.type == "rss" &&
-        !targetFolder.isSpecialFolder(Ci.nsMsgFolderFlags.Trash, true) &&
-        event.dataTransfer.items.length == 1 &&
-        FeedUtils.getFeedUriFromDataTransfer(event.dataTransfer)
-      ) {
-        return;
-      }
-      event.dataTransfer.dropEffect = "link";
     } else {
       return;
     }
 
@@ -3343,8 +3343,17 @@ var folderPane = {
         console.warn(
           "Bug 1896531. Copy and move for multiselection is only partially supported and it might fail."
         );
       }
+    } else if (
+      targetFolder.server.type == "rss" &&
+      (types.includes("text/x-moz-url-data") ||
+        types.includes("text/x-moz-url"))
+    ) {
+      // This is a potential rss feed. A link image as well as link text url
+      // should be handled; try to extract a url from non moz apps as well.
+      const feedURI = FeedUtils.getFeedUriFromDataTransfer(event.dataTransfer);
+      FeedUtils.subscribeToFeed(feedURI.spec, targetFolder);
     } else if (types.includes("application/x-moz-file")) {
       for (let i = 0; i < event.dataTransfer.mozItemCount; i++) {
         const extFile = event.dataTransfer
           .mozGetDataAt("application/x-moz-file", i)
@@ -3378,16 +3387,8 @@ var folderPane = {
       }
       setTimeout(() => {
         folderTree.swapSelection(rows);
       });
-    } else if (
-      types.includes("text/x-moz-url-data") ||
-      types.includes("text/x-moz-url")
-    ) {
-      // This is a potential rss feed. A link image as well as link text url
-      // should be handled; try to extract a url from non moz apps as well.
-      const feedURI = FeedUtils.getFeedUriFromDataTransfer(event.dataTransfer);
-      FeedUtils.subscribeToFeed(feedURI.spec, targetFolder);
     }
 
     event.preventDefault();
   },
